### PR TITLE
chore(sidebar): Make hover in the sidebar work like in the connection list COMPASS-6112

### PR DESCRIPTION
**Pull request by lerouxb**
**Before:**

https://user-images.githubusercontent.com/69737/191012063-6d1051bf-a7fe-4651-85e1-f6b3571d0d0c.mov

(just realised that in main hover background colour is overriding active background color. oops)

**After:**

https://user-images.githubusercontent.com/69737/191011405-314640a1-e9f5-47ca-879c-e40991420b08.mov

### CODE REVIEW

File: packages/compass-databases-navigation/src/collection-item.tsx [hunk 1]
```diff
@@ -130,22 +132,29 @@ export const CollectionItem: React.FunctionComponent<
...
-      </ItemLabel>
-      <ItemActionControls<Actions>
-        className={collectionActions}
-        onAction={onAction}
-        data-testid="sidebar-collection-item-actions"
-        iconSize="small"
-        isVisible={isActive || isHovered}
-        actions={actions}
-      ></ItemActionControls>
+      <ItemWrapper numIcons={actions.length}>
```
Comment (gribnoysup) [inline packages/compass-databases-navigation/src/collection-item.tsx:138]: I think relying on actions length here gives weird results sometimes, for example collections will have from 2 to 4 actions available, always would show one icon here as this behavior is based on the `collapseToMenuThreshold` prop of the `ItemActionControls`, not on the actual value of the items

```suggestion
      <ItemWrapper numIcons={1}>
```

File: packages/compass-databases-navigation/src/database-item.tsx [hunk 2]
```diff
@@ -150,29 +152,36 @@ export const DatabaseItem: React.FunctionComponent<
...
-          className={databaseActions}
-          onAction={onAction}
-          isVisible={isActive || isHovered}
-          data-testid="sidebar-database-item-actions"
-          collapseToMenuThreshold={3}
-          iconSize="small"
-          actions={actions}
-        ></ItemActionControls>
-      )}
+      <ItemWrapper numIcons={actions.length}>
```
Comment (gribnoysup) [inline packages/compass-databases-navigation/src/database-item.tsx:158]: Ditto here

```suggestion
      <ItemWrapper numIcons={Math.max(2, actions.length)}>
```

File: packages/compass-databases-navigation/src/tree-item.tsx [hunk 3]
```diff
@@ -58,28 +59,39 @@ export function useDefaultAction<T>(
...
+  },
+
+  ':hover .item-background': {
+    display: 'block',
     backgroundColor: 'var(--item-bg-color-hover)',
   },
 
+  '.item-action-controls:hover + .item-background': {
+    display: 'none',
+  },
```
Comment (gribnoysup) [inline packages/compass-databases-navigation/src/tree-item.tsx:77]: One super minor thing, so totally feel free to ignore: something that tripped me up during a review is that emotion seem to have different way to concatenate selectors based on whether it is a pseudo selector or not (so like `:hover` will be `.as6rf:hover`, but `.item` will become `.as6rf .item`) and so maybe just to make this explicit we can use `&` there

```suggestion
  '& .item-action-controls': {
    marginLeft: 'auto',
    marginRight: spacing[1],
  },

  '&:hover .item-background': {
    display: 'block',
    backgroundColor: 'var(--item-bg-color-hover)',
  },

  '& .item-action-controls:hover + .item-background': {
    display: 'none',
  },
```

**Review by gribnoysup — APPROVED** [review]
Awesome, I really like how this works now

File: packages/compass-databases-navigation/src/tree-item.tsx (hunk 3 above)
Comment (lerouxb) [reply packages/compass-databases-navigation/src/tree-item.tsx:77]: Seems sensible. I was surprised it worked this way actually, but saw it done elsewhere which is why I did it this way. I'll add the &

**Pull request merged**